src/__tests__/metrics.test.ts

```diff
@@ -8,21 +8,52 @@ const params = {
 }
 
 describe('metrics', () => {
-  test.each([[counter, Counter], [histogram, Histogram], [gauge, Gauge], [summary, Summary]])(
-    '%p passes correct params to %p',
-    (a, b) => {
-      a(params)
-      expect(b).toHaveBeenCalledWith(params)
-    }
-  )
+  test.each([
+    [counter, Counter],
+    [histogram, Histogram],
+    [gauge, Gauge],
+    [summary, Summary]
+  ])('%p passes correct params to %p', (a, b) => {
+    // @ts-expect-error (ts2339)
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    register.getSingleMetric.mockReturnValue(undefined)
+    a(params)
+    expect(b).toHaveBeenCalledWith(params)
+  })
+
+  test.each([[counter], [histogram], [gauge], [summary]])('prevents to register duplicate metrics, method=%p', (method) => {
+    // @ts-expect-error (ts2339)
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    register.getSingleMetric.mockReturnValue(undefined)
+    const result1 = method(params)
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(register.getSingleMetric).toHaveBeenCalledTimes(1)
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(register.getSingleMetric).toHaveBeenLastCalledWith(params.name)
+
+    // @ts-expect-error (ts2339)
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    register.getSingleMetric.mockReturnValue(result1)
+    const result2 = method(params)
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(register.getSingleMetric).toHaveBeenCalledTimes(2)
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(register.getSingleMetric).toHaveBeenLastCalledWith(params.name)
+    expect(result1).toEqual(result2)
+  })
```

Comment:
Does it make sense to split these too?

Reply:
I don't think so, because the test says "prevents to register duplicate metrics", so we need to call it twice

Reply:
Ok I get it now. I think the test logic can be simplified. What if we setup 2 calls and then assert that the `register.getSingleMetric` was only called once?

Im not trying to be picky but I feel like this test can be simplified. Wdyt? @jlongo-encora

Reply:
@ignatospadov
When calling twice `counter(params)` with the same `params.name`, `register.getSingleMetric` will be called twice. We do need to ensure the return objects are the same.